bin/fetch-topics

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+URL=https://raw.githubusercontent.com/exercism/problem-specifications/master/TOPICS.txt
+curl -s --location $URL > TOPICS.txt
```

Comment:
This will work on Linux/BSD (and maybe Mac?), but we'll need a platform-independent solution.

I got curious while writing this comment, and ended up writing a solution before even post it :stuck_out_tongue:. The code below saves the file, but actually, this code could (and maybe should) just be built into the script so that it doesn't need to save and then read the file.

```python
try:
    # Python 3
    from urllib.request import urlopen
except ImportError:
    # Python 2
    from urllib2 import urlopen


URL = 'https://raw.githubusercontent.com/exercism/problem-specifications/master/TOPICS.txt'

topics_file = urlopen(URL).read()

with open("TOPICS.txt", "wb") as f:
    f.write(topics_file)
```

It might be worth making the script fail gracefully if there's an error fetching the topics list and make it just not check the topics list? I feel like it would be a useful thing from a user point of view (the script would work when offline), but I'm not sure whether we be risking Travis-CI not checking topics if there's a (presumably unlikely) network error?

Reply:
@N-Parsons 

I have no argument against a platform-independent solution, but if curl is not good enough for this, we should change it for `fetch-configlet` too.

My argument for writing topics to a file is for use cases where `config_helper.py` is run multiple times (probably just the `edit` and `lint` commands); performing a network action is must more time consuming than simply reading from a file, if the file is already available.

Regarding offline availability, I'm not entirely sure that's a practical use case. Sure, you can write code offline, but you'll need a network connection eventually to push commits.

There is another option regarding `TOPICS.txt`. We create our own curated track TOPICS.txt that is strictly a subset of the global topics list. We could (optionally) remove topics that are not applicable in Python (such as `lenses`). I'm not particularly in favor of this option, but it would eliminate another network dependency.

Reply:
@cmccandless, I hadn't seen that `fetch-configlet` was doing that. It seems to be an exercism-wide script, so I'm a little confused. I'm away at the moment and so don't currently have access to Windows to check, but I will take a look at it early next week. I'll create an issue for this later though so that it's not forgotten.

I'm not sure how often the `TOPICS.txt` file will be updated now that it's mostly settled in. My only concern about storing a file is that it could become out-of-date. There are a few different ways to handle this, but I'm not sure what would be best. I guess that Travis-CI would always download a fresh copy, so errors shouldn't get through even without an up-to-date copy.

I wouldn't be completely opposed to keeping our own subset `TOPICS.txt` and then considering upstream additions and optionally adding them. It would add a little more work for us, but I could easily write a scripted check that I can leave running on my server to give me a heads up about changes (perhaps even changes to `problem-specifications` in general). I could make them go to a public Trello board, or I could even make it raise issues here automatically.

TL;DR:
* I'll take a look at `fetch-configlet` to see if it needs some work.
* I think it's ok to store the file using the Python script.
* I'll write something to watch `problem-specifications` for us when I get chance.

Reply:
It most definitely doesn't work on Windows. Command Prompt can't run bash script files, and running it under MinGW-bash (git-bash) results in the Linux executable being downloaded.

Reply:
@N-Parsons I've implemented platform-independent file downloading (with zip/tar support, as appropriate) in `fetch.py`. I also ported `fetch-configlet` to Python and confirmed correct functionality on Windows. Travis passed, so Linux is also functioning correctly.